### PR TITLE
test: Refactor EVM unit test fixture

### PR DESCRIPTION
**Pull request by chfast**
Continuation of #135.

### CODE REVIEW

File: test/unittests/evm_fixture.hpp [hunk 1]
```diff
@@ -35,46 +35,55 @@
 class evm : public testing::Test
 {
 protected:
+    /// The VM handle.
     evmc::vm& vm;
-    evmc_revision rev = EVMC_BYZANTIUM;  // Byzantium by default. TODO: Add alias evmc::revision.
-    evmc_message msg = {};               // TODO: Add evmc::message with default constructor.
-    evmc::result result{{}};  // TODO: Add default constructor to evmc::result, update code here.
+
+    /// The EVM revision for unit tests execution. Byzantium by default.
```
Comment (halfalicious) [inline test/unittests/evm_fixture.hpp]: (nit) unit tests execution -> unit test execution

File: test/unittests/evm_fixture.hpp [hunk 2]
```diff
@@ -35,46 +35,55 @@
...
+    /// TODO: Add alias evmc::revision.
+    evmc_revision rev = EVMC_BYZANTIUM;
+
+    /// The message to be executed by an unit test (with execute() method).
```
Comment (halfalicious) [inline test/unittests/evm_fixture.hpp]: (Nit) An -> a

File: test/unittests/evm_fixture.hpp [hunk 3]
```diff
@@ -35,46 +35,55 @@
...
+    /// TODO: Add evmc::message with default constructor.
```
Comment (halfalicious) [inline test/unittests/evm_fixture.hpp]: msg isn’t invoking the default evmc_message ctor?

File: test/unittests/evm_fixture.hpp [hunk 4]
```diff
@@ -35,46 +35,55 @@
...
     int64_t gas_used = 0;
 
     MockedHost host;
 
     evm() noexcept : vm{get_vm()} {}
 
-    /// Wrapper for evmone::execute. The result will be in the .result field.
-    void execute(int64_t gas, bytes_view code, std::string_view input_hex = {}) noexcept
+
+    /// Executes given code.
```
Comment (halfalicious) [inline test/unittests/evm_fixture.hpp]: (Nit) Executes given code -> executes the supplied code (or “executes the given code”)

File: test/unittests/evm_fixture.hpp [hunk 5]
```diff
@@ -35,46 +35,55 @@
...
+    ///
+    /// @param gas        The gas limit for execution.
```
Comment (halfalicious) [inline test/unittests/evm_fixture.hpp]: I think “the” can be removed from these arg comments

File: test/unittests/evm_test.cpp [hunk 6]
```diff
@@ -30,13 +29,14 @@ TEST_F(evm, push_implicit_data)
     // This test executes 1 byte code with a push instruction without the push data following.
     // Unfortunately, there is no result we could observe other than program crash.
 
-    // Allocate 1 byte code on heap to allow detecting memory issues by tools like valgrind.
-    auto code = std::make_unique<uint8_t>();
+    // Create long code prefix to enforce it to be placed on heap. Invalid heap access is easier
```
Comment (halfalicious) [inline test/unittests/evm_test.cpp]: (Nit) Enforce -> force

File: test/unittests/evm_fixture.hpp (hunk 3 above)
Comment (chfast) [reply test/unittests/evm_fixture.hpp]: It's a C struct.

File: test/unittests/evm_fixture.hpp (hunk 5 above)
Comment (chfast) [reply test/unittests/evm_fixture.hpp]: I followed the LLVM documentation. If both versions are fine, let's keep the existing one.

**Pull request merged**